### PR TITLE
Check for bad passwords using zxcvbn

### PR DESCRIPTION
**Pull request by rogeruiz**
[Bad passwords are based on a score of <=2](https://github.com/dropbox/zxcvbn/blob/63bd219723de9052b5db1b1b288ab7c7ca573df9/src/feedback.coffee#L16). This is what the library uses
to determine whether or not to supply feedback or warnings, so I figured
it was good enough. Maybe we should make it configurable? But, meh.

See Jira here:
https://cm-jira.usa.gov/browse/CG-866

### CODE REVIEW

**Comment by rogeruiz** [comment]
I'd like to write a test where we make sure that the redirect occurs if we supply a bad password to ensure that the logic of `< 3` is working. But I don't have time for that today. I'll get to it on Monday.

**Comment by rogeruiz** [comment]
Goes with the work on 18F/uaa-customized-boshrelease#21

File: setup.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
         'email_validator==1.0.1',
         'talisman==0.1.0',
         'redis==2.10.5',
+        'zxcvbn-python=4.4.27',
```
Comment (jmcarp) [inline setup.py]: Should be `zxcvbn-python==4.4.27`

File: uaaextras/tests.py [hunk 2]
```diff
@@ -546,17 +537,51 @@ def test_post_change_password_bad_too_simple(self, render_template, uaac, zxcvbn
...
+            )
+
+        uaac().change_password.assert_called_with(
+            'example',
+            'fdsa',
+            'correct horse battery staple'
+        )
+        render_template.assert_called_with('password_changed.html')
+        flash.assert_not_called()
+        assert uaac().change_password.called
```
Comment (jmcarp) [inline uaaextras/tests.py]: This assertion is redundant with the `assert_called_with` above. We already asserted that the method was called with specific arguments, so we don't need to assert again that it was called.

File: uaaextras/tests.py [hunk 3]
```diff
@@ -507,36 +508,26 @@ def test_post_change_password_good(self, render_template, uaac):
...
+        assert not uaac().change_password.called
 
-    @patch('uaaextras.webapp.flash')
-    @patch('uaaextras.webapp.zxcvbn')
     @patch('uaaextras.webapp.UAAClient')
     @patch('uaaextras.webapp.render_template')
-    def test_post_change_password_bad_too_simple(self, render_template, uaac, zxcvbn, flash):
-        """When a POST request is made to /forgot-password
-           and the user exists, send_email is called
+    def test_post_change_password_bad_repeat(self, render_template, uaac):
```
Comment (jmcarp) [inline uaaextras/tests.py]: From the name of the test, I expected that we would test the behavior of the `repeat_password` field--meaning we would test the case where `new_password` and `repeat_password` are different. This actually tests the case where `old_password` and `new_password` aren't different. Could you rename the test to describe what it does?

Comment (rogeruiz) [reply uaaextras/tests.py]: I addressed this in `a38a094` and yeah @timothy-spencer I was thinking the same thing when I mentioned `new` !== `repeat`.

**Comment by pburkholder** [comment]
Works for me in staging

**Pull request merged**